### ISSUE TITLE
VLC full screen controls missing when desktop has 2 or more nodes

### ISSUE THREAD

**Issue by phisch** [baskerville/bspwm#903]
bspwm: 0.9.5
vlc: 3.0.6

When opening a video with vlc, where vlc is the only node on the desktop, moving the mouse will get vlc to fade in (depending on compositing manager of course) the control bar.

When doing the same on a desktop that has more than just the vlc node, moving the mouse will not reveal the control bar.

I think this is a bspwm issue because when i run the latest git build, even when vlc is the only node, the control bar will not show up. So this has gotten worse since the last stable.

I am sure this has worked flawlessly in the past, so something must have sneaked in.

**Comment by phisch** [comment]
Compiling bspwm 0.9.3 tag, the issue does not exist in 0.9.3, must have been introduced later.

**Comment by phisch** [comment]
At 0f10b9219d17ab00932555537a161303d69040dd it kind of works as long as the full-screen window keeps the focus, if you focus another window, and then focus the full-screen vlc again, it fails to show the control bar again.

One commit later at a5acc9a24eabad5d722af0754a1030b917ccb9b1 it fails to show the control bar when there are more than one nodes on the desktop that contains the full screen vlc.

**Comment by phisch** [comment]
Okay, bisected this issue all the way from 0.9.3 to master

0.9.3 <- fine
a5acc9a24eabad5d722af0754a1030b917ccb9b1 <- broken
... <- broken
a94093b4685b4e31b605271c69b4b3e5c1c9c1b3 <- working
2c2faeb959bc38f9e25dce695d60b230fd4462a8 <- working
3e7f820afe86623c758165972b7651d19903860b <- completely broken

So a5acc9a24eabad5d722af0754a1030b917ccb9b1 made it so the control panel would not show when more than one node is on the desktop. a94093b4685b4e31b605271c69b4b3e5c1c9c1b3 completely fixed the issue. And 3e7f820afe86623c758165972b7651d19903860b made it so the control panel never shows up, so this commit is the latest one that completely changed this behavior.

**Comment by dieggsy** [comment]
I was experiencing this - where are the VLC fullscreen controls? I realized they still show up, but as a floating window in the `normal` layer, whereas fullscreen windows are `above` by default. Floating windows have precedence (show up on top) _within the same layer_, so as a workaround for this, I use an external rule that does `layer=above border=off` for windows of instance `vlc` with title `vlc` (which I think is just the controller). This seems to make it behave how i want.

**Comment by baskerville** [comment]
I was going to suggest that: as of 3e7f820, if a fullscreen window is in layer *L1* and a floating window in layer *L2*, then the floating window will be above the fullscreen window if and only if *L2 > L1*.

I'm not sure whether or not the behaviour of VLC is in accordance with the EWMH standard. If it isn't, you might want to report the problem upstream.

**Comment by dieggsy** [comment]
@baskerville to clarify, the floating window will be above the fullscreen window if L2 = L1 too, right? Cause I'm pretty sure this is how my workaround is working.

**Comment by baskerville** [comment]
@dieggsy No, as of 3e7f820, within the same layer, full windows are above floating windows.

**Comment by dieggsy** [comment]
@baskerville oh, fair. Then I must have misremembered why my rule works - the fullscreen window must be in the `normal` layer - moving the floating window to `above` gets it to show up. Thanks for the clarification.
